# Housing Market

### Introduction:

This time we will create our own dataset with fictional numbers to describe a house market. As we are going to create random data don't try to reason of the numbers.

### Step 1. Import the necessary libraries

In [2]:
import pandas as pd
import numpy as np

### Step 2. Create 3 differents Series, each of length 100, as follows:
1. The first a random number from 1 to 4
2. The second a random number from 1 to 3
3. The third a random number from 10,000 to 30,000

In [3]:
np.random.seed(42)

series1 = pd.Series(np.random.randint(1, 5, size=100))
series2 = pd.Series(np.random.randint(1, 4, size=100))
series3 = pd.Series(np.random.randint(10000, 30001, size=100))


In [4]:
series1.head()

0    3
1    4
2    1
3    3
4    3
dtype: int64

In [5]:
series2.head()

0    3
1    2
2    2
3    2
4    2
dtype: int64

In [7]:
series3.head()

0    23931
1    13627
2    26157
3    20173
4    28047
dtype: int64

### Step 3. Let's create a DataFrame by joinning the Series by column

In [8]:
df = pd.concat([series1, series2, series3], axis=1)
df.head()

0  1      2
0  3  3  23931
1  4  2  13627
2  1  2  26157
3  3  2  20173
4  3  2  28047

### Step 4. Change the name of the columns to bedrs, bathrs, price_sqr_meter

In [9]:
df.rename(columns={'Column1': 'bedrs', 'Column2': 'bathrs', 'Column3': 'price_sqr_meter'}, inplace=True)
df.head()

0  1      2
0  3  3  23931
1  4  2  13627
2  1  2  26157
3  3  2  20173
4  3  2  28047

### Step 5. Create a one column DataFrame with the values of the 3 Series and assign it to 'bigcolumn'

In [10]:
bigcolumn = pd.concat([series1, series2, series3], axis=0)
df_bigcolumn = pd.DataFrame({'bigcolumn': bigcolumn})
df_bigcolumn.head()

bigcolumn
0          3
1          4
2          1
3          3
4          3

### Step 6. Oops, it seems it is going only until index 99. Is it true?

In [11]:
bigcolumn = pd.concat([series1, series2, series3], axis=0, ignore_index=True)
df_bigcolumn = pd.DataFrame({'bigcolumn': bigcolumn})
df_bigcolumn

bigcolumn
0            3
1            4
2            1
3            3
4            3
..         ...
295      11154
296      14499
297      16295
298      22183
299      22874

[300 rows x 1 columns]

### Step 7. Reindex the DataFrame so it goes from 0 to 299

In [12]:
df_bigcolumn_reindexed = df_bigcolumn.reset_index(drop=True)
df_bigcolumn_reindexed

bigcolumn
0            3
1            4
2            1
3            3
4            3
..         ...
295      11154
296      14499
297      16295
298      22183
299      22874

[300 rows x 1 columns]